Hangman/Hangman.py

```diff
@@ -0,0 +1,64 @@
+from Board import Board
+from Generate_word import GenerateWord
+from User_Player import UserPlayer
+
+
+def is_alphabet(letter) -> bool:
+    if len(letter) > 1:
+        return False
+    return letter in "abcdefghijklmnopqrstuvwxyz"
+
+
+class Hangman:
+    def __init__(self):
+        self.board = Board()
+        self.userPlayer = UserPlayer()
+
+    def ask_for_game(self) -> bool:
+        want_game = self.userPlayer.want_game()
+        return want_game
+
+    def start_game(self) -> None:
+        self.board.set_word(GenerateWord.get_random_word())
+        self.run_game()
+
+    def run_game(self) -> None:
+        if self.board.has_chances():
+            self.guessing()
+            if self.board.guessed_it():
+                print("you have guessed the word ", end="\n\n")
+                return
+            else:
+                self.run_game()
+        else:
+            print("no chances left")
+            print("word is ", self.board.get_word(), end="\n\n")
+            return
+
+    def guessing(self) -> None:
+        print("You have {} chance ".format(self.board.get_chance()))
+        letter = self.userPlayer.guess_letter()
+
+        if not is_alphabet(letter):
+            print("It is not  an alphabet try an alphabet")
+        elif self.board.word_has(letter):
+            print("right guess")
+            for i in range(len(self.board.get_word())):
+                if self.board.equal_word_at_pos(i, letter):
+                    self.board.set_guessed_word_at(i, letter)
+        else:
+            print("wrong guess")
+            if not self.board.guessed_letters_has(letter):
```

Comment:
your board class offers a lot of dis integrated APIs that the caller is supposed to call in correct sequence . drawback of this flaw is
1. having a separate board class is not achieving anything
2. its very easy for board state to get in a bad state if caller messes up the order or checks
3.   caller has a lot of responsibility and board is not providing a correct set of functions to maintain its state that overwrites the whole point of having a separate board class.

All the decrement /increment logic and conditions should be in board class itself with specific functions which are exposed to caller to call

Reply:
like there could be a function guess which would do all the stuff in board itself.
board.guess(letter)